src/AspNetCore/Inc/inprocessapplication.h

```diff
@@ -54,6 +56,28 @@ class IN_PROCESS_APPLICATION : public APPLICATION
             VOID
         );
 
+    REQUEST_NOTIFICATION_STATUS
+    OnAsyncCompletion(
+        IHttpContext*           pHttpContext,
+        DWORD                   cbCompletion,
+        HRESULT                 hrCompletionStatus
+    );
+
+
+    HRESULT
+    GetInProcessStoredContext(
+        IHttpContext* pHttpContext,
+        IN_PROCESS_STORED_CONTEXT** ppInProcessStoredContext
+    );
+
+    VOID
+    SetWebsocketsEnabled(
```

Comment:
you don't need this function. we have a global variable g_fWebSocketSupported which is set to true for win8 and above only

Reply:
This is state stored on the application to say whether the current stream is websockets or not. If I register a callback to IHttpResponse2::WriteEntityChunks(), will that trigger the OnAsyncCompletion method?